extensions/cloud/portability-cloud-google/src/main/java/org/dataportabilityproject/cloud/google/GoogleJobStore.java

```diff
@@ -142,6 +118,11 @@ public void updateJob(UUID jobId, PortabilityJob job) throws IOException {
         throw new IOException("Could not find record for jobId: " + jobId);
       }
 
+      if (previousState != null) {
+        PortabilityJob previousJob = PortabilityJob.fromMap(getProperties(previousEntity));
```

Comment:
For a future PR: Maybe if we json serialize the job, we won't have to go from entity -> map -> job object.